### ISSUE TITLE
Schema not compiling for a value class with Integer value

### ISSUE THREAD

**Issue by bjornbak** [andyglow/scala-jsonschema#25]
I try to create a schema for
```
case class PixelConfiguration(
                               name: String,
                               comments: List[String] = List(),
                               pixel: Option[Pixel] = None,
                               publisherId: Set[PublisherId] = Set(),
                               forwardPercent: Option[Int] = None
                             ) extends Model with Validates[PixelConfiguration]
```
where 
```
case class PublisherId(val value: Int) extends AnyVal with IntField with Validates[PublisherId] 
```
with
```
val schemaJson: Json = Json.schema[List[PixelConfiguration]].asCirce(Version.Draft07(id = s"http://example.com/pixel-config-$version.json$timestamp"))
```
but get a type mismatch:
```
[error]  found   : json.Schema.integer.type
[error]  required: json.Schema[com.liveintent.dwh.common.fields.PublisherId]
[error]     val schemaJson: Json = Json.schema[List[PixelConfiguration]].asCirce(Version.Draft07(id = s"http://example.com/pixel-config-$version.json$timestamp"))
```

I have tried to follow the answers you gave in https://github.com/andyglow/scala-jsonschema/issues/19 and https://github.com/andyglow/scala-jsonschema/issues/22 but can't convince to Json.schema to treat as a list of PublisherId instead of Integers.

How do I force/workaround this?

**Comment by andyglow** [comment]
That's because `PublisherId` is actually a value class.. `AnyVal`. Right?
That is interesting. Let me think how to resolve that. I a meanwhile, don't you want to go without `AnyVal`?

**Comment by bjornbak** [comment]
Unfortunately I can't as it is part of a larger framework serializing to S3, Spark and more.
I just need to create json based config files for a subsystem.

**Comment by andyglow** [comment]
fixed in 0.5.1
